src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -22,114 +22,119 @@
+	"repair tool of player"
+})
 @Since("1.0")
 public class EffHealth extends Effect {
 
 	static {
 		Skript.registerEffect(EffHealth.class,
-			"damage %livingentities/itemtypes% by %number% [heart[s]] [with fake cause %-damagecause%]",
+			"damage %livingentities/itemtypes/slots% by %number% [heart[s]] [with fake cause %-damagecause%]",
 			"heal %livingentities% [by %-number% [heart[s]]]",
-			"repair %itemtypes% [by %-number%]");
+			"repair %itemtypes/slots% [by %-number%]");
 	}
 
-	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<?> damageables;
 	@Nullable
-	private Expression<Number> damage;
-	private boolean heal = false;
+	private Expression<Number> amount;
+	private boolean isHealing, isRepairing;
 
-	@SuppressWarnings({"unchecked", "null"})
+	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		if (matchedPattern == 0 && exprs[2] != null)
 			Skript.warning("The fake damage cause extension of this effect has no functionality, " +
 				"and will be removed in the future");
 
-		damageables = exprs[0];
-		if (!LivingEntity.class.isAssignableFrom(damageables.getReturnType())) {
-			if (!ChangerUtils.acceptsChange(damageables, ChangeMode.SET, ItemType.class)) {
-				Skript.error(damageables + " cannot be changed, thus it cannot be damaged or repaired.");
-				return false;
-			}
-		}
-		damage = (Expression<Number>) exprs[1];
-		heal = matchedPattern >= 1;
-
+		this.damageables = exprs[0];
+		this.isHealing = matchedPattern >= 1;
+		this.isRepairing = matchedPattern == 2;
+		this.amount = (Expression<Number>) exprs[1];
 		return true;
 	}
 
 	@Override
-	public void execute(Event e) {
-		double damage = 0;
-		if (this.damage != null) {
-			Number number = this.damage.getSingle(e);
-			if (number == null)
+	protected void execute(Event event) {
+		double amount = 0;
+		if (this.amount != null) {
+			Number amountPostCheck = this.amount.getSingle(event);
+			if (amountPostCheck == null)
 				return;
-			damage = number.doubleValue();
+			amount = amountPostCheck.doubleValue();
 		}
-		Object[] array = damageables.getArray(e);
-		Object[] newArray = new Object[array.length];
 
-		boolean requiresChange = false;
-		for (int i = 0; i < array.length; i++) {
-			Object value = array[i];
-			if (value instanceof ItemType) {
-				ItemType itemType = (ItemType) value;
-				ItemStack itemStack = itemType.getRandom();
+		for (Object obj : this.damageables.getArray(event)) {
+			if (obj instanceof ItemType) {
+				ItemType itemType = (ItemType) obj;
+
+				if (this.amount == null) {
+					ItemUtils.setDamage(itemType, 0);
+				} else {
+					ItemUtils.setDamage(itemType, (int) Math2.fit(0, (ItemUtils.getDamage(itemType) + (isHealing ? -amount : amount)), itemType.getMaterial().getMaxDurability()));
+				}
```

Comment:
Due to 1.20.5 changes this is now technically bugable due to custom durability component. Forgot all about that. 
In addition `Damageable` ItemMeta is available on all items as well.

Reply:
Do you think this is worth delaying this addition, or should we get it in 2.6.5 and worry about the 1.20.5 changes later?

Reply:
Looking over it, is appears spigot has just thrown the the `#setMaxDamage` method into `Damageable`, I can quickly see how a test build of this interacts but as paper still has no stable release only betas I likely cannot push it for awhile longer.